presto-main/src/main/java/io/prestosql/split/PageSourceManager.java

```diff
@@ -47,18 +49,34 @@ public void removeConnectorPageSourceProvider(CatalogName catalogName)
     }
 
     @Override
-    public ConnectorPageSource createPageSource(Session session, Split split, TableHandle table, List<ColumnHandle> columns)
+    public ConnectorPageSource createPageSource(Session session, Split split, TableHandle table, List<ColumnHandle> columns, Supplier<TupleDomain<ColumnHandle>> dynamicFilter)
     {
         requireNonNull(columns, "columns is null");
         checkArgument(split.getCatalogName().equals(table.getCatalogName()), "mismatched split and table");
         CatalogName catalogName = split.getCatalogName();
 
-        return getPageSourceProvider(catalogName).createPageSource(
-                table.getTransaction(),
-                session.toConnectorSession(catalogName),
-                split.getConnectorSplit(),
-                table.getConnectorHandle(),
-                columns);
+        ConnectorPageSourceProvider provider = getPageSourceProvider(catalogName);
+        TupleDomain<ColumnHandle> constraint = TupleDomain.all();
+        if (dynamicFilter != null) {
+            constraint = dynamicFilter.get(); // should not block
```

Comment:
@rzeyde-varada @sopel39 
Thank you for your node local dynamic filtering work. It's really helpful.

> `// should not block`

Though I didn't have the context before, I've spent some time trying to understand the code. From my (coarse) understanding, the `ScanFilterAndProjectOperator` doesn't wait for the dynamic filter to be collected. So it is possible that the `constraint` we get here is `null` even if the dynamic filter is being actively collected on the hash build side. Is this understanding correct?

Reply:
You're welcome :)

> So it is possible that the `constraint` we get here is `null` even if the dynamic filter is being actively collected on the hash build side. 

Yes, this is indeed possible.

Reply:
Thanks for the clarification!